mypy/semanal.py

```diff
@@ -406,7 +407,8 @@ def _visit_func_def(self, defn: FuncDef) -> None:
                         # Redefinition. Conditional redefinition is okay.
                         n = self.locals[-1][defn.name()].node
                         if not self.set_original_def(n, defn):
-                            self.name_already_defined(defn.name(), defn)
+                            self.name_already_defined(defn.name(), defn,
+                                                      self.locals[-1][defn.name()])
```

Comment:
It looks like you missed one more case: global function definition, see `else: # Top-level function` below. You will likely need to patch also `self.check_no_global` in the same way you do above for `self.name_already defined`

Reply:
I think `self.check_no_global` already does the right thing?
https://github.com/python/mypy/blob/7eebb8f9d433488f7deff7b4cba14dc3a9650743/mypy/semanal.py#L3225

Reply:
OK, if the previous context is always correct in `check_no_global`, then no action is required here.